composer.json

```diff
@@ -11,13 +11,13 @@
   "license": "MIT",
   "require": {
     "php": "~8.1.0 || ~8.2.0 || ~8.3.0",
-    "phpunit/phpunit": "^10.2.6"
+    "phpunit/phpunit": "^10.2 || ^11.0"
   },
   "require-dev": {
     "friendsofphp/php-cs-fixer": "^3.15.1",
     "roave/security-advisories": "dev-latest",
-    "vimeo/psalm": "^5.8.0",
-    "rector/rector": "^0.18.0"
+    "rector/rector": "^0.18.0",
+    "psalm/phar": "^5.24"
```

Comment:
Psalm and PHPUnit 11 conflict with PHP-Parser v4 and v5.

```
% composer require -W 'phpunit/phpunit:^11.0'
./composer.json has been updated
Running composer update phpunit/phpunit --with-all-dependencies
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - phpunit/php-code-coverage[11.0.0, ..., 11.0.3] require nikic/php-parser ^5.0 -> found nikic/php-parser[v5.0.0, v5.0.1, v5.0.2] but these were not loaded, likely because it conflicts with another require.
    - phpunit/phpunit[11.0.0, ..., 11.1.3] require phpunit/php-code-coverage ^11.0 -> satisfiable by phpunit/php-code-coverage[11.0.0, 11.0.1, 11.0.2, 11.0.3].
    - Root composer.json requires phpunit/phpunit ^11.0 -> satisfiable by phpunit/phpunit[11.0.0, ..., 11.1.3].
```

```
% composer why nikic/php-parser
phpunit/php-code-coverage 10.1.14 requires  nikic/php-parser (^4.18 || ^5.0)
sebastian/complexity      3.2.0   requires  nikic/php-parser (^4.18 || ^5.0)
sebastian/lines-of-code   2.0.2   requires  nikic/php-parser (^4.18 || ^5.0)
vimeo/psalm               5.23.1  requires  nikic/php-parser (^4.16)
vimeo/psalm               5.23.1  conflicts nikic/php-parser (4.17.0)
```

The [psalm/phar](https://github.com/psalm/phar) is a good option to avoid dependency conflicts.